pkg/data/data.go

```diff
@@ -0,0 +1,94 @@
+package data
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+)
+
+type Recipient struct {
+	PhoneNumber   string
+	Quantity      string
+	Unit          string
+	Validity      string
+	IsPromoBundle string
+	MetaData      interface{}
+}
+
+/*
+DataRequest represents the structure for the request
+body to be sent to Africa's Talking api endpoint
+*/
+type DataRequest struct {
+	Username    string
+	ProductName string
+	Recipients  []Recipient
+}
+
+type Entry struct {
+	PhoneNumber   string `json:"phoneNumber"`
+	Provider      string `json:"provider"`
+	Status        string `json:"status"`
+	TransactionId string `json:"transactionId"`
+	Value         string `json:"value"`
+}
+
+type DataResponse struct {
+	Entries []Entry `json:"entries"`
+}
+
+/*
+Client represents the http client for communicating with Africa's
+Talking Api
+*/
+type Client struct {
+	ApiKey    string       //Api Key provided by Africa's Talking
+	Username  string       // Your Africa's Talking application username
+	IsSandbox bool         // Is Sandbox specifies whether to use sandbox or live environment
+	Client    *http.Client //
+}
+
+func setHeaders(request *http.Request, apiKey string) {
+	request.Header.Set("apiKey", apiKey)
+	request.Header.Set("accept", "application/json")
+	request.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+}
+
+func (c *Client) SendMobileData(request *DataRequest) (DataResponse, error) {
```

Comment:
The naming seems redundant:

```go
data.SendMobileData() // Redundant

data.Send() // Concise  
```

```go
request *DataRequest // Redundant

request *Request // Concise
```

Given that this entire package is soley for sending Data, there isn't a need to include the word `Data`